src/emu/luaprinter.h

```diff
@@ -0,0 +1,116 @@
+// copyright-holder: Golden Child
+/***************************************************************************
+
+    luaprinter.h
+
+    Utility class for printing functions
+
+***************************************************************************/
+
+#ifndef MAME_LUAPRINTER_H
+#define MAME_LUAPRINTER_H
+
+#pragma once
+
+#include "emu.h"
+#include "png.h"
+#include "emuopts.h"
+
+class luaprinter {
+
+public:
+	luaprinter(device_t& thisdevice, std::string dummyname);
+	~luaprinter() { lp_removefromprinterlist(); }
+
+	device_t *m_lp_mydevice;
+	static std::vector<luaprinter *> luaprinterlist;
```

Comment:
You shouldn’t be doing this.  Use an iterator to walk the device graph and find devices that derive from your mixin.